CMakeLists.txt

```diff
@@ -253,31 +253,192 @@ endif()
+      set(icon_font_binding_regex "void register_constants_${icon_font_base}\\(py::module &m\\) {")
+      file(STRINGS "${icon_font_pybind}" icon_font_binding_found REGEX "${icon_font_binding_regex}")
+      if (NOT icon_font_binding_found)
+        message(
+          FATAL_ERROR "Python font binding found searching file [${icon_font_pybind}]. "
+                      "Regex searched for: ${icon_font_binding_regex}"
+        )
+      endif()
+
+      # Copy the generated file so that #include "python.h" works
+      file(
+        COPY "${icon_font_pybind}"
+        DESTINATION "${CMAKE_CURRENT_SOURCE_DIR}/python"
+      )
+      # Add this additional source file to be compiled into the python bindings
+      list(
+        APPEND LIBNANOGUI_PYTHON_EXTRA_SOURCE
+        "${CMAKE_CURRENT_SOURCE_DIR}/python/constants_${icon_font_base}.cpp"
+      )
+
+      # Create the configuration variables for python/main.cpp, which amount to
+      # declaring an extern void register_constants_fontname(py::module &m) at
+      # the top, and then later calling that method.
+      set(
+        NANOGUI_EXTRA_ICON_REGISTRY_DEFINITIONS
+        "${NANOGUI_EXTRA_ICON_REGISTRY_DEFINITIONS}\nextern void register_constants_${icon_font_base}(py::module &m);"
+      )
+      set(
+        NANOGUI_EXTRA_ICON_REGISTRY_INVOCATIONS
+        "${NANOGUI_EXTRA_ICON_REGISTRY_INVOCATIONS}\n    register_constants_${icon_font_base}(m);"
+      )
+    endif()
+
+    # If everything was successful up to this point, then the preliminary
+    # requirements for embedding an additional icon font have been met.
+    # Appending to NANOGUI_EXTRA_RESOURCES will enable bin2c below to finalize
+    # the inclusion of this font in NanoGUI.
+    list(APPEND NANOGUI_EXTRA_RESOURCES "${icon_font}")
+  endforeach()
+endif()
 
 # Run simple cmake converter to put font files into the data segment
 
 # Glob up resource files
 file(GLOB resources "${CMAKE_CURRENT_SOURCE_DIR}/resources/*.ttf")
 
+# Parent projects can set this list to include additional fonts:
+#
+#     list(APPEND NANOGUI_EXTRA_RESOURCES "${CMAKE_CURRENT_SOURCE_DIR}/path/to/font.ttf")
+#
+# before executing add_subdirectory(path/to/nanogui).  True type (.ttf) and Open Type
+# (.otf) font files are supported.
+if (NANOGUI_EXTRA_RESOURCES)
+  # Open type fonts (OTF) are only supported if freetype can be found.
+  find_package(Freetype QUIET)
+  set(NANOGUI_USE_FREETYPE FALSE) # Only use when needed (.otf file requested)
+  foreach (font ${NANOGUI_EXTRA_RESOURCES})
+    get_filename_component(font_ext "${font}" EXT)
+    if (NOT "${font_ext}" STREQUAL ".ttf" AND NOT "${font_ext}" STREQUAL ".otf")
+      message(FATAL_ERROR "NanoGUI can only embed true type and open type fonts.  [${font}] does not end with '.ttf' or '.otf'.")
+    endif()
+
+    if ("${font_ext}" STREQUAL ".otf")
+      if (NOT FREETYPE_FOUND)
+        message(
+          WARNING "[${font}] is an Open Type Font (.otf), but Freetype was not "
```

Comment:
I wasn't sure whether to make this a `WARNING` or a `FATAL_ERROR`.  I chose `WARNING` because it appears freetype may be cumbersome on Windows.  I was planning on just incorporating in the docs how to accommodate this.  The idea is basically that `-1` will be returned when *loading* the font, so at that point you can either fail out like `Theme` does with an `std::runtime_error`, or you could choose to fallback.